Source/Core/Core/HW/SI/SI_DeviceGCPedometer.cpp

```diff
@@ -0,0 +1,156 @@
+// Copyright 2018 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "Core/HW/SI/SI_DeviceGCPedometer.h"
+
+#include <cstring>
+
+#include "Common/ChunkFile.h"
+#include "Common/CommonTypes.h"
+#include "Common/Logging/Log.h"
+#include "Common/MsgHandler.h"
+#include "Common/Swap.h"
+#include "Core/HW/SI/SI_Device.h"
+
+namespace SerialInterface
+{
+// Gamecube Pedometer aka Inro-Kun
+CSIDevice_GCPedometer::CSIDevice_GCPedometer(SIDevices device, int device_number)
+    : ISIDevice(device, device_number)
+{
+}
+
+int CSIDevice_GCPedometer::RunBuffer(u8* buffer, int length)
+{
+  // For debug logging only
+  ISIDevice::RunBuffer(buffer, length);
+
+  // Read the command
+  EBufferCommands command = static_cast<EBufferCommands>(buffer[0]);
+
+  // Process command
+  switch (command)
+  {
+  case CMD_RESET:
+  {
+    // Return joybus ID for buffer commands
+    u32 id = Common::swap32(SI_GC_PEDOMETER);
+    std::memcpy(buffer, &id, sizeof(id));
+    break;
+  }
+
+  // Return data from pedometer
+  case CMD_DATA:
+  {
+    // Pedometer expects 0x51 bytes from buffer
+    if (length == 0x51)
+    {
+      // Clear existing data on SI buffer for pedometer response
+      for(int i = 0; i < length; i++)
```

Comment:
This could just be `std::memset(buffer, 0, length)` if I'm not mistaken.

Reply:
or std::fill_n

Reply:
Wasn't sure what convention to use. Any preference between memset or fill_n?

Reply:
I'd prefer std::fill_n as it's type-safe and you pass it the number of elements rather than the number of bytes.